### ISSUE TITLE
Networking page should recommend a fix for race condition

### ISSUE THREAD

**Issue by brandon-rhodes** [docker/docker#5940]
My recent update to the `networking.md` page omits an important issue — on which I need advice from more experienced Docker developers before I can address it.

The question: if the user opts for `--net=none` and wants to set up networking for a container manually with their own shell script, how should they handle the race condition of not being able to set up networking until the container is established, and the race condition of maybe having the process inside of the container need to wait until the network appears?

When Docker itself is configuring the networking, it can simply do these three steps sequentially and assure that they do not overlap:

1.  Create the network namespace for the new container.
2.  Add interfaces, rename them, configure them, and bring them up.
3.  `exec()` the actual in-container process that the user asked for, which will look around and see a well-configured network port already up and available.

But when the user wants to take charge of Step 2, it is not clear how their shell script (or other automated command) can safely make sure it happens after Step 1 (because firing off too early means networking commands that die because the namespace does not exist yet) but before Step 3 (so the process they are starting does not try binding to interfaces that do not exist yet; irrelevant, I suppose, for users who simply bind to `0.0.0.0` and therefore maybe not a real issue).

The [approach that I am current taking](https://github.com/brandon-rhodes/fopnp/blob/m/playground/play.sh) sits in a `while` loop in the shell waiting for `docker inspect` to be able to (a) see the container and (b) fetch its PID successfully (since the only way I know of to generate its network namespace is by building the path `/proc/$pid/ns/net` manually) before I proceed with my network commands. Should this be recommended for other users?

If `docker run` took a `--setup script.sh` option which was guaranteed to run on the Docker host in between creation of the container and the kickoff of the user executable inside of the container, then the whole problem would go away — but if there is a way of resolving the race condition without resorting to a new option, it would be great to solve this without Docker having to become any more complicated.

So: is there a recommended approach that I can mention in `networking.md` (and use in my own projects) that helps a setup script wait until a container is up and running with a network namespace that is ready to be plumbed and configured? Thanks for any guidance! In response I will be happy to update `networking.md` to codify any information you can provide.

**Comment by jfrazelle** [comment]
closing as duplicate of https://github.com/docker/docker/issues/6549
